### PR TITLE
Store required libs on model object

### LINKED ISSUE #354: Updated required libs functionality breaks with solo models
Solo models break with functionality introduced in #351 because their config is never populated with `required_libs`.

The error looks like this
```python
Traceback (most recent call last):
  File "/g/data3/hh5/public/apps/miniconda3/envs/analysis3-23.04/bin/payu-run", line 10, in <module>
    sys.exit(runscript())
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-23.04/lib/python3.9/site-packages/payu/subcommands/run_cmd.py", line 132, in runscript
    expt.run()
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-23.04/lib/python3.9/site-packages/payu/experiment.py", line 525, in run
    envmod.lib_update(
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-23.04/lib/python3.9/site-packages/payu/envmod.py", line 101, in lib_update
    for lib_filename, lib_path in required_libs.items():
AttributeError: 'NoneType' object has no attribute 'items'
```
and occurs for configs with a single solo model, like [MOM6-CICE6](https://github.com/COSIMA/MOM6-CICE6/blob/main/config.yaml).

Each of the sub-models config has the correct information injected here:

https://github.com/payu-org/payu/blob/master/payu/experiment.py#L143

but this not done if there is a solo model.

### PR DESCRIPTION
**Pull request by jo-basevi**
Store information of required libs per binary on the Model object if there exists an executable path. 

Should close #354 and #355

### CODE REVIEW

**Review by aidanheerdegen — APPROVED** [review]
Looks great!

We should run the `MOM6-CICE6` example that was erroring for Ezhil, instructions on how to do this are here:

https://github.com/COSIMA/access-om3/wiki/Quick-start

And also do a quick run of one of the ACCESS-OM2 configs, e.g.

https://github.com/COSIMA/1deg_jra55_ryf

but change the runtime (called `restart_period`)

https://github.com/COSIMA/1deg_jra55_ryf/blob/master/accessom2.nml#L19

to something like

    restart_period = 0, 0, 10800

which will run for 2 time steps (3 hours)

File: payu/experiment.py [hunk 1]
```diff
@@ -138,17 +138,9 @@ def init_models(self):
...
-            sm['required_libs'] = required_libs(sm['exe'])
-
-        solo_model = self.config.get('model')
-        if not solo_model:
-            sys.exit('payu: error: Unknown model configuration.')
-
         submodel_config = dict((f, self.config[f]) for f in model_fields
                                if f in self.config)
-        submodel_config['name'] = solo_model
+        submodel_config['name'] = self.model_name
```
[the remark this answers is not in the document]
Comment (jo-basevi) [reply payu/experiment.py:143]: Yes, the `assert self.model_name` would raise an assertion error and exit if `self.config.get('model')` was false, so I am sure that the `sys.exit()` under the `if not solo_model:` was unreachable.

[the remark this answers is not in the document]
Comment (aidanheerdegen) [reply payu/experiment.py:143]: Thanks.

**Comment by jo-basevi** [comment]
Ok, I let it run for a bit with MOM6-CICE6 (I wasn't sure how to do a short run of it so I used `qdel` after it ran for awhile). Though as required_libs are set during setup so I could check that after `payu setup`, the required_libs field were set as expected. 
Everything still works fine with the ACCESS-OM2 configs

**Comment by aidanheerdegen** [comment]
Feel free to merge if you are happy it is good to go @jo-basevi

**Comment by aidanheerdegen** [comment]
I've also done some checking because Ezhil was reporting odd unrelated behaviour (that is a different issue), but it all seems to work well with single model and multi-model configs.

**Comment by aidanheerdegen** [comment]
I just thought, we probably want to check for storage paths in the `use` path.

That would be added here

https://github.com/payu-org/payu/blob/105fdc0930a43ff0802c9091ebbc6d8c85d6fa40/payu/schedulers/pbs.py#L112

Ideally we might want to look in the paths of the `required_libs`, but that isn't known until after the run is submitted to PBS, so not available.

**Comment by jo-basevi** [comment]
Ok, I've added 'module use' paths to the list to search for storage paths and added it to the tests.

**Comment by aidanheerdegen** [comment]
Great, thanks! Good to merge I think.

**Pull request merged**